### PR TITLE
feature(ui-ux): `balances` - sort tokens based on USD value

### LINKED ISSUE #2089: `balances` - sort tokens based on USD value
#### What would you like to be added:
On the balances page, we would like to sort it based on USD value. 
#### Why is this needed:
Users will see first tokens with amounts on the balances page.

### PR DESCRIPTION
**Pull request by chloezxyy**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #2089

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [x] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx [hunk 1]
```diff
@@ -108,6 +108,8 @@ export function BalancesScreen ({ navigation }: Props): JSX.Element {
       totalUSDValue: new BigNumber(0),
       dstTokens: []
     })
+    // to sort USD values in descending order
+    const sortedUsdAmt = dstTokens.sort((a, b) => parseInt(b.usdAmount.toFixed(8)) - parseInt(a.usdAmount.toFixed(8)))
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx:112]: image

`parseInt` would remove the decimal points, could you check on how you can sort it with BigNumber funcs?

File: mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx [hunk 2]
```diff
@@ -108,6 +108,8 @@ export function BalancesScreen ({ navigation }: Props): JSX.Element {
       totalUSDValue: new BigNumber(0),
       dstTokens: []
     })
+    // to sort USD values in descending order
+    const sortedUsdAmt = dstTokens.sort((a, b) => parseFloat(b.usdAmount.toFixed(8)) - parseFloat(a.usdAmount.toFixed(8)))
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx:112]: Just chain this with the `.map`. `dstTokens.sort.map`

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Comment by kyleleow** [comment]
Can we wrap `tokens.reduce()` with `useCallback()` at https://github.com/DeFiCh/wallet/blob/1ae6ea699a9fba981b0574dc04d0920d53be2db6/mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx#L82 for performance sake?

File: mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts [hunk 3]
```diff
@@ -545,3 +545,20 @@ context('Wallet - Balances - USD Value', () => {
...
+  before(function () {
+    cy.createEmptyWallet(true)
+    cy.sendDFItoWallet().wait(3000)
+    cy.getByTestID('header_settings').click()
+    cy.getByTestID('bottom_tab_balances').click()
+  })
+
+  it('should display LTC on top of ETH after topping up more LTC', function () {
+    cy.sendTokenToWallet(['ETH', 'LTC']).wait(3000)
+    // dLTC will be displayed at the top of the card on first topup
```
Comment (kyleleow) [inline mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts:559]:
```suggestion
    // dETH will be displayed at the top of the card on first topup
```

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**